### PR TITLE
fix: dont round off floats while formatting

### LINKED ISSUE #678: Formatting bug on ISO8601
Hi @jhpratt continuing our discussion from #674. I have created a repo to reproduce this bug [here](https://github.com/dracarys18/reproduce_time_bug). You can try to run it and check it yourself.

```bash
cargo r | tee test.txt
cat test.txt | rg ":60."
```

This looks like this happens only when the TimePrecision is 3 here.

```rust
    const FORMAT_CONFIG: EncodedConfig = Config::DEFAULT
        .set_time_precision(TimePrecision::Second {
            decimal_digits: NonZeroU8::new(3),
        })
        .encode();
```

It looks like it's rounding off the time to 60 seconds and it's happening every minute. If I change the precision to 6 I can see it's adjusting to nano second

### PR DESCRIPTION
**Pull request by dracarys18**
Fixes #678

### CODE REVIEW

File: time/src/formatting/mod.rs [hunk 1]
```diff
@@ -40,6 +41,31 @@ const WEEKDAY_NAMES: [&[u8]; 7] = [
     b"Sunday",
 ];
 
+/// Wrapper struct for f64 with custom Display formatter
+#[derive(Debug, Clone, Copy)]
+pub struct FloatNum(f64);
```
Comment (dracarys18) [inline time/src/formatting/mod.rs:46]: Made this public for tests

**Comment by dracarys18** [comment]
@jhpratt Any updates on this? We are kind of blocked on this

**Comment by jhpratt** [comment]
Sorry, this fell through the cracks. I'll take a look soon, though I'll likely want to make some changes myself to avoid exposing anything unnecessarily.

**Comment by dracarys18** [comment]
> Sorry, this fell through the cracks. I'll take a look soon, though I'll likely want to make some changes myself to avoid exposing anything unnecessarily.

Sure thanks let me know that the changes are

**Pull request merged**